### ISSUE TITLE
Support for compression

### ISSUE THREAD

**Issue by bridger** [vapor/websocket-kit#55]
I'm curious if compression support has been considered in this package. I'd be interested in using it in my server-side app. 

Kitura developed this package for websocket compression which is also using NIO's websockets. Maybe it could be adapted to work with Vapor? https://github.com/IBM-Swift/Kitura-WebSocket-Compression

**Comment by tanner0101** [comment]
It looks like the only dependency that package has is SwiftNIO, so it would be possible to use that if WebSocketKit exposed some way to add additional handlers. 

I wonder if this is something SwiftNIO would consider adopting. That would make it really easy for WebSocketKit to allow configuring compression like we do for HTTP.

**Comment by bridger** [comment]
Yeah, I wasn’t sure if this is better at the NIO level or the library level that uses NIO. I’ll make another issue to ask there.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>
